pandas/core/groupby/groupby.py

```diff
@@ -4767,8 +4767,20 @@ def rank(
             result = self._python_apply_general(
                 f, self._selected_obj, is_transform=True
             )
+            if method != "average" and not pct:
+                temp = result
+                temp = temp.convert_dtypes()
```

Comment:
This is calling Series.rank / DataFrame.rank, why is it necessary to convert the dtype here?

Reply:
Maybe I've misinterpreted the issue description. In this example given by the issue reporter:

```
In [5]: df.rank()
Out[5]: 
     a    b
0  2.0  1.0
1  2.0  2.0
2  2.0  3.0
3  4.5  4.0
4  4.5  5.0
```

I thought that it was requested for column b in this case (and in analogous cases) to return int64, i.e.:

 ```
In [5]: df.rank()
Out[5]: 
     a    b
0  2.0   1
1  2.0   2
2  2.0   3
3  4.5   4
4  4.5   5
```

Was I wrong?

Reply:
Also, I used convert_dtypes() because I read in the documentation in a use example that it transforms float64 into int64 when relevant (e.g. 5.0 = 5), and I couldn't think of possible errors other than maybe NaN becoming <NA>.